pkg/function-manager/handlers.go

```diff
@@ -456,6 +454,8 @@ func (h *Handlers) updateFunction(params fnstore.UpdateFunctionParams, principal
 		})
 	}
 
+	id := uuid.NewV4().String()
+	e.BaseEntity.ID = id
```

Comment:
This does not seem right, why do we do that?

Reply:
This was to force the creation of a new function in the FaaS. This id is not used internally, but it is used as a key for creating new functions.

Reply:
Can you add a comment to this in the code? I would never guess it myself

Reply:
Wait, this is not a good idea.  We should not be changing the ID of our entities.  The correct way to handle this is to create a new field called `FaasID` or `FunctionID` which maps to the FaaS generated ID/name.  We should not change the ID of our entities, it isn't an "update" anymore.

Reply:
So, what you have described for FaasID is exactly how we are using ID. ID is not used internally by dispatch. We uniquely identify functions by a key which is made up of the type of the entity and the name. I can easily add this extra field if you feel this is confusing as implemented.

Reply:
ID is supposed to represent the key of an entity which is unique across time (unlike name).  Even though we don't expose it directly today, I don't want us to change the meaning.  An entity with a different ID is a NEW entity, not an updated one.  I expect the dispatch function entity to continue to have the same ID across updates, even if the FaaS function does not.